packages/hawtio/src/plugins/shared/workspace.ts

```diff
@@ -44,11 +62,13 @@ class Workspace implements IWorkspace {
 
   private async loadTree(): Promise<MBeanTree> {
     if (!(await userService.isLogin())) {
-      throw new Error('User needs to have logged in to use workspace')
+      this.addError(new Error('User needs to have logged in to use workspace'))
+      return MBeanTree.createEmpty(pluginName)
```

Comment:
This changes the behaviour when the user is not logged in yet. Is it really safe?  Can't we just reserve the original behaviour while recording the error in the workspace?